src/charm.py

```diff
@@ -368,6 +388,26 @@ def _on_collect_unit_status(self, e: CollectStatusEvent):
             else:
                 e.add_status(ActiveStatus())
 
+    def _on_nginx_pebble_ready(self, _) -> None:
+        self.nginx.configure_pebble_layer(tls=self.tls_available)
+
+    def _on_nginx_prometheus_exporter_pebble_ready(self, _) -> None:
+        self.nginx_prometheus_exporter.configure_pebble_layer()
+
+    def _on_event(self, event) -> None:
+        """A set of common configuration actions that should happen on every event."""
+        if isinstance(event, CollectStatusEvent):
+            return
+        # plan layers
+        self.nginx.configure_pebble_layer(tls=self.tls_available)
+        self.nginx_prometheus_exporter.configure_pebble_layer()
+        # configure ingress
+        self._configure_ingress()
+        # update cluster relations
+        self._update_tempo_cluster()
```

Comment:
I think this causes that a new tempo config gets pushed to the workers. The workers then reconcile and every time this happens, tempo will be down until the new config is reflected and tempo is up again. It causes unneeded `cluster-changed` event at least every 5 mins (with update-status event)

Reply:
Not sure as I think the assumption was that it won't propagate to the workers if everything is the same: https://github.com/canonical/tempo-coordinator-k8s-operator/blob/main/src/charm.py#L493

Will test it out further tomorrow.

Reply:
Doesn't seem like I'm getting the extra events?

```
ubuntu@charm-dev:~/tempo-coordinator-k8s-operator$ juju show-status-log tempo-worker-k8s/0
Time                        Type       Status     Message
10 Jul 2024 18:46:28+02:00  juju-unit  error      hook failed: "start"
10 Jul 2024 18:46:33+02:00  juju-unit  executing  running start hook
10 Jul 2024 18:46:36+02:00  juju-unit  executing  running tempo-pebble-ready hook
10 Jul 2024 18:47:40+02:00  juju-unit  idle       
10 Jul 2024 19:02:58+02:00  juju-unit  executing  running tempo-cluster-relation-changed hook
10 Jul 2024 19:03:10+02:00  juju-unit  idle       
10 Jul 2024 19:20:55+02:00  juju-unit  executing  running tempo-cluster-relation-departed hook for tempo-coordinator-k8s/0
10 Jul 2024 19:20:56+02:00  workload   active     (all roles) ready.
10 Jul 2024 19:20:56+02:00  juju-unit  executing  running tempo-cluster-relation-broken hook
10 Jul 2024 19:20:57+02:00  workload   blocked    Missing tempo-cluster relation to a tempo-coordinator charm
10 Jul 2024 19:20:57+02:00  juju-unit  idle       
10 Jul 2024 19:22:39+02:00  juju-unit  executing  running tempo-cluster-relation-created hook
10 Jul 2024 19:22:40+02:00  juju-unit  idle       
10 Jul 2024 19:23:05+02:00  juju-unit  executing  running tempo-cluster-relation-joined hook for tempo-coordinator-k8s/0
10 Jul 2024 19:23:06+02:00  juju-unit  executing  running tempo-cluster-relation-changed hook for tempo-coordinator-k8s/0
10 Jul 2024 19:23:06+02:00  workload   waiting    Waiting for coordinator to publish a tempo config
10 Jul 2024 19:23:06+02:00  juju-unit  idle       
10 Jul 2024 19:23:16+02:00  juju-unit  executing  running tempo-cluster-relation-changed hook
10 Jul 2024 19:23:32+02:00  juju-unit  idle       
11 Jul 2024 11:45:24+02:00  workload   active     (all roles) ready.
```

Reply:
maybe its sth in my setup.